kraken-build/src/kraken/std/cargo/tasks/cargo_sync_config_task.py

```diff
@@ -36,10 +38,33 @@ def __init__(self, name: str, project: Project) -> None:
 
     def get_file_contents(self, file: Path) -> str | bytes:
         content = tomli.loads(file.read_text()) if not self.replace.get() and file.exists() else {}
-        content.setdefault("registry", {})["global-credential-providers"] = ["cargo:token"]
+        content.setdefault("registry", {})["global-credential-providers"] = [
+            "cargo:libsecret",
+            "cargo:macos-keychain",
+            "cargo:wincred",
+            "cargo:token",
```

Comment:
This is confusing because I get the feeling that the behavior changes every day. But from my tests today, and from [Cargo's source code](https://github.com/rust-lang/cargo/blob/master/src/cargo/util/auth/mod.rs#L110), my understanding is that the credential providers should be provided in _reverse_ order of preference.

I have amended the code and added a comment.